xs/src/libslic3r/Model.hpp

```diff
@@ -28,221 +28,529 @@ typedef std::vector<ModelObject*> ModelObjectPtrs;
+    ///< to new volumes before adding them to this object in order to preserve alignment
+    ///< when user expects that.
+
     // these should be private but we need to expose them via XS until all methods are ported
     BoundingBoxf3 _bounding_box;
     bool _bounding_box_valid;
-    
+
+    ///  Get the owning parent Model.
+    /// \return parent Model* pointer to the owner Model
     Model* get_model() const { return this->model; };
-    
+
+    /// Add a new ModelVolume to the current ModelObject.
+    /// \param mesh TriangularMesh
+    /// \return ModelVolume* pointer to the new volume
     ModelVolume* add_volume(const TriangleMesh &mesh);
+
+    /// Add a new ModelVolume to the current ModelObject.
+    /// \param volume the ModelVolume object to be copied
+    /// \return ModelVolume* pointer to the new volume
     ModelVolume* add_volume(const ModelVolume &volume);
+
+    /// Delete a ModelVolume object.
+    /// \param idx size_t the index of the ModelVolume to be deleted
     void delete_volume(size_t idx);
+
+    /// Delete all ModelVolumes in the
     void clear_volumes();
 
+    /// Add a new ModelInstance to the current ModelObject.
+    /// \return ModelInstance* a pointer to the new instance
     ModelInstance* add_instance();
+
+    /// Add a new ModelInstance to the current ModelObject.
+    /// \param instance the ModelInstance to be copied
+    /// \return ModelInstance* a pointer to the new instance
     ModelInstance* add_instance(const ModelInstance &instance);
+
+    /// Delete a ModelInstance.
+    /// \param idx size_t the index of the ModelInstance to be deleted
     void delete_instance(size_t idx);
+
+    /// Delete the last created ModelInstance object.
     void delete_last_instance();
+
+    /// Delete all ModelInstance objects found in the current ModelObject.
     void clear_instances();
 
+    /// Get the bounding box of the *transformed* instances.
     BoundingBoxf3 bounding_box();
+
+    /// Invalidate the bounding box in the current ModelObject.
     void invalidate_bounding_box();
 
+    /// Repair all TriangleMesh objects found in each ModelVolume.
     void repair();
+
+    /// Flatten all volumes and instances into a single mesh and applying all the ModelInstances transformations.
     TriangleMesh mesh() const;
+
+    /// Flatten all volumes and instances into a single mesh.
     TriangleMesh raw_mesh() const;
+
+    /// Get the raw bounding box
+    /// \return BoundingBoxf3
     BoundingBoxf3 raw_bounding_box() const;
+
+    /// Get the bounding box of the *transformed* given instance.
+    /// \param instance_idx size_t the index of the ModelInstance in the ModelInstance vector
+    /// \return BoundingBoxf3 the bounding box at the given index
     BoundingBoxf3 instance_bounding_box(size_t instance_idx) const;
+
+    /// Align the current ModelObject to ground by translating in the z axis the needed units.
     void align_to_ground();
+
+    /// Center the current ModelObject to origin.
```

Comment:
Specify whether this happens by changing the instances offsets or by translating volumes.